src/denoise.cpp

```diff
@@ -211,6 +213,48 @@ static void check_init() {
   common.init = 1;
 }
 
+static void post_filtering(float* g, const float* Ey) {
+    // Envelope Postfiltering
+    int i = 0;
+    float E0 = 0;
+    float E1 = 0;
+    float E_div = 0;
+    float g_w[NB_BANDS] = { 0 };
+    float G = 0.0f;
+
+    // warped gain
+    for (i = 0; i<NB_BANDS; i++) {
+      g_w[i] = g[i]*sinf(M_PI/2*g[i]);
+    }
+
+    // total energy of the enhanced signal
+    for (i = 0; i<NB_BANDS; i++) {
+      E0 += g[i]*Ey[i];
+    }
+
+    // total energy when using the warped gain
+    for (i = 0; i<NB_BANDS; i++) {
+      E1 += g_w[i]*Ey[i];
+    }
+
+    // global gain compensation heuristic
+    E_div = E0/(E1+1e-6f);
+    G = sqrtf(
+          ((1+ENVELOPE_POSTFILTERING_BETA)*E_div)/(1+ENVELOPE_POSTFILTERING_BETA*(E_div*E_div))
+        );
+
+    // Scaling the final signal for the frame by G
+    for (i = 0; i<NB_BANDS; i++) {
+      float e = G*g_w[i];
+      if (e > 1.0f) {
+        e = 1.0f;
+      } else if (e < 0.0f) {
+        e = 0.0f;
+      }
+      g[i] = e;
```

Comment:
I think for compensating clean band sometimes gain should be more than 1.0
so the code should be changed like below.
```
g[i] = G*g_w[i];
```